src/components/user/CreateTargetForm.js

```diff
@@ -0,0 +1,124 @@
+import React, { PureComponent } from 'react';
+import { connect } from 'react-redux';
+import { func, string, bool, array } from 'prop-types';
+import { Field, reduxForm } from 'redux-form/immutable';
+import {
+  injectIntl,
+  intlShape,
+  defineMessages,
+  FormattedMessage
+} from 'react-intl';
+
+import Loading from 'components/common/Loading';
+import Input from 'components/common/Input';
+import CustomSelect from 'components/common/CustomSelect';
+import { validations, createTarget } from 'utils/constraints';
+import createTargetIcon from './../../assets/targetGroup/group.png';
+import smileIcon from './../../assets/smilies/smilies@3x.png';
+
+const messages = defineMessages({
+  specifyArea: { id: 'target.form.specifyArea' },
+  targetTitle: { id: 'target.form.targetTitle' },
+  selectTopic: { id: 'target.form.selectTopic' },
+  placeChooseTitle: { id: 'target.form.placeHolderTitle' },
+  placeHolderTopics: { id: 'target.form.placeHolderTopics' }
+});
+
+export class CreateTargetForm extends PureComponent {
+  static propTypes = {
+    topicsList: array,
+    handleSubmit: func.isRequired,
+    intl: intlShape.isRequired,
+    submitting: bool.isRequired,
+    error: string,
+  }
+
+  state = {
+    topics: []
+  };
+
+  componentDidMount() {
+    this.mapTopicsToArray();
+  }
+
+  mapTopicsToArray() {
+    const { topicsList = [] } = this.props;
+    const topics = topicsList.map(item =>
```

Comment:
Can you use destructuring here with `item`?

Reply:
I have just changed this.